ape_tenderly/provider.py

```diff
@@ -37,3 +39,41 @@ def connect(self):
 
     def disconnect(self):
         self._web3 = None
+
+
+class TenderlyGatewayProvider(Web3Provider, UpstreamProvider):
+    """
+    A web3 provider using an HTTP connection to Tenderly's RPC nodes.
+
+    Docs: https://docs.tenderly.co/web3-gateway/web3-gateway
+    """
+
+    @property
+    def uri(self) -> str:
+        project_id = os.environ.get("TENDERLY_GATEWAY_ACCESS_KEY")
+        assert project_id is not None
+        network_name = self.network.name
+        return f"https://{network_name}.gateway.tenderly.co/{project_id}"
+
+    @property
+    def connection_str(self) -> str:
+        return self.uri
+
+    def connect(self):
+        self._web3 = Web3(HTTPProvider(self.uri))
```

Comment:
do we want to verify `is_connected()` here?

Reply:
It pretty quuickly calls `Web3.eth.chain_id` so that validates the connection